hack/tests/sanity-check.sh

```diff
@@ -8,5 +8,21 @@ go fmt ./...
 ./hack/check_license.sh
 ./hack/check_error_log_msg_format.sh
 
+# Make sure that test/test-framework is updated according to the current source code
+make install
+cd test/test-framework
+
+# Create files in the mock test/test-framework just for we are able to exec the commands
+cat > build/Dockerfile
+cat > go.mod
+
+GO111MODULE=on operator-sdk generate k8s
+GO111MODULE=on operator-sdk generate openapi
```

Comment:
> Add sanity check to ensure that the CR/CRD's and k8s,openapi files will be always be updated according to the release.

Just to be clear the manifests in `test/test-framework` won't really be updated by just regenerating the manifests here in the sanity-check stage.
All this is doing is checking if the commands `generate k8s` and `generate openapi` run without errors against the existing types in `test/test-framework`.

Even if the manifests do get updated here(due to some upstream change in controller-tools) this won't persist for the test stages that actually use the `test/test-framework` manifests since they run in a separate job.
https://github.com/openshift/release/blob/f34ec4ae1234aa156df1802c67963a35aa8f1941/ci-operator/config/operator-framework/operator-sdk/operator-framework-operator-sdk-master.yaml#L130-L133

`subcommand-scorecard.sh`
https://github.com/operator-framework/operator-sdk/blob/e03cde67f916af43b9e15edf6b976ec9894b23e6/ci/tests/subcommand-scorecard.sh#L5-L6

`subcommand.sh`
https://github.com/operator-framework/operator-sdk/blob/e03cde67f916af43b9e15edf6b976ec9894b23e6/hack/tests/subcommand.sh#L10-L12

If we want to run those tests with updated `test/test-framework` manifests every time then `generate k8s` and `generate openapi` should be moved to the two scripts above.

/cc @estroz @AlexNPavel to double check how `test/test-framework` should be updated.

Reply:
Hi @hasbro17, 

The sanity will not update the files. It will return an error if after we run the operator-sdk gen commands the test-framework has changes which mean that it is outdated.  

Is it make sense now?

Reply:
Hi @hasbro17 I changed the title/description after our chat. Sorry if I did not make it clear for the first time and lead you to have a diff understanding here. Really tks for your review.

Reply:
@camilamacedo86 I think it would be helpful context for the PR if you could list all of the ways that the test framework is currently used in existing tests, down to specifics (e.g “the scorecard subcommand test uses the test framework as an example operator to verify correct functionality of scorecard features”)

Without seeing that list, it is difficult to assess the value of keeping the test framework’s generated code up-to-date, especially given that this PR would require the sanity test to start building and running the operator-ask binary.

Along the lines of what @hasbro17 was suggesting, it may be more lightweight to leave the generated code out of the test framework in the repo and to instead generate it on the fly in the tests where it is necessary (and where we already have an operator-ask binary).

Reply:
Hi @joelanford,


> @camilamacedo86 I think it would be helpful context for the PR if you could list all of the ways that the test framework is currently used in existing tests, down to specifics (e.g “the scorecard subcommand test uses the test framework as an example operator to verify correct functionality of scorecard features”)

I will check the places which are using the test-framework and add in the first comment. However, would really make sense do any test with any gen artefact that is no longer generated in that way? How to keep the files outdated could help us to ensure the quality of the code/changes performed in any circumstance? 

> Without seeing that list, it is difficult to assess the value of keeping the test framework’s generated code up-to-date, especially given that this PR would require the sanity test to start building and running the operator-ask binary

I do not see what harm it could cause. The make install runs in less than 1 second.

> Along the lines of what @hasbro17 was suggesting, it may be more lightweight to leave the generated code out of the test framework in the repo and to instead generate it on the fly in the tests where it is necessary (and where we already have an operator-ask binary).

What is the value to add a check that is not called in the CI? I mean, it will surely get obsulate and never used/checked.

Reply:
Thanks @camilamacedo86. That's helpful. Another angle to look at this is to ask the question, "which of the tests require the files that are created or updated by the `generate` subcommands?".

AFAICT, those files are:
1. test/test-framework/deploy/crds/cache.example.com_memcachedrs_crd.yaml
2. test/test-framework/deploy/crds/cache.example.com_memcacheds_crd.yaml
3. test/test-framework/pkg/apis/cache/v1alpha1/zz_generated.deepcopy.go
4. test/test-framework/pkg/apis/cache/v1alpha1/zz_generated.openapi.go

We know that none of the tests require `zz_generated.openapi.go` since we're just now adding it and all tests have passed before.

For the other three files, which test's require those? To figure this out, I deleted those files from the repo and ran CI to see what would fail. The tests that fail without these files are:
1. The [subcommand tests](https://travis-ci.org/joelanford/operator-sdk/jobs/609422129#L806)
2. The [sanity test](https://travis-ci.org/joelanford/operator-sdk/jobs/609422130#L755)

The [Go, Helm, and Ansible e2e tests passed](https://travis-ci.org/joelanford/operator-sdk/builds/609422125) without those files.

Since the sanity test (specifically `go vet ./...`) requires these files, we need to have them checked in or we need to generate them prior to running `go vet`. We recently added another task to the sanity test to [generate CLI docs](https://github.com/operator-framework/operator-sdk/blob/5488645f7bccfb7480b83c65e21a04b438c6c489/hack/tests/sanity-check.sh#L10), so I'm coming around to doing this.

What do you think of reorganizing things like the following?

1. Put these generation commands (minus the `make install` command) into a `./hack/generate/gen-test-framework.sh` script.
2. Call `./hack/generate/gen-test-framework.sh` from the sanity test.
3. Add a `gen-test-framework` target to the Makefile that depends in `install` and calls the above script.
4. Rename `cli-doc` to `gen-cli-doc` in Makefile
5. Add `generate` target that depends on `gen-cli-doc` and `gen-test-framework`
6. Move `./hack/doc/*` to `./hack/generate/` and update references to these scripts in the Makefile and sanity test script.

Thoughts @theishshah and @hasbro17?

Also @asmacdo wondering if you have any thoughts on this organization as it relates to the doc website generation?

Reply:
Really thank you for your help and inputs. I will change its implementation in order to follow your suggestion.

Reply:
Done 👍

Reply:
Done 👍